File: implementations/java-springboot/examples/src/main/java/com/example/accountservice/config/AwsProperties.java
package com.example.accountservice.config;

import jakarta.validation.constraints.NotBlank;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.validation.annotation.Validated;

@ConfigurationProperties(prefix = "aws")
@Validated
public record AwsProperties(
        @NotBlank String region, String endpointUrl, String accessKeyId, String secretAccessKey) {}

File: implementations/kotlin-springboot/examples/src/main/kotlin/com/example/accountservice/config/AwsProperties.kt
package com.example.accountservice.config

import jakarta.validation.constraints.NotBlank
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.validation.annotation.Validated

@Validated
@ConfigurationProperties(prefix = "aws")
data class AwsProperties(
    @field:NotBlank
    val region: String,
    val endpointUrl: String = "",
    val accessKeyId: String = "",
    val secretAccessKey: String = "",
)
